packages/features/ee/workflows/lib/reminders/providers/twilioProvider.ts

```diff
@@ -284,11 +284,16 @@ export async function getCountryCodeForNumber(phoneNumber: string) {
   return countryCode;
 }
 
-export async function getPriceForSMS(smsSid: string) {
+export async function getMessageInfo(smsSid: string) {
   const twilio = createTwilioClient();
   const message = await twilio.messages(smsSid).fetch();
-  if (message.price == null || message.price === "null") return null;
-  return Math.abs(parseFloat(message.price));
+
+  const price =
+    message.price != null && message.price !== "null" ? Math.abs(parseFloat(message.price)) : null;
+  const numSegments =
+    message.numSegments != null && message.numSegments !== "null" ? parseInt(message.numSegments) : null;
```

Comment:
```suggestion
const price = message.price?.toString() !== "null" ? Math.abs(parseFloat(message.price)) : null;
const numSegments = message.numSegments?.toString() !== "null" ? parseInt(message.numSegments) : null;
```

Reply:
This can cause `NaN` values. Tested the twilio webhooks response and it's always `null` or the price in string format, so this should work: https://github.com/calcom/cal.com/pull/21771/commits/463bb856024cfb052ceef2523beed832761667c1#diff-340b8e1acef76e2d190f6c466fdc0d2d8a3499b5ead6623ef9f609a5dc455458R290